sentry_sdk/utils.py

```diff
@@ -658,19 +664,56 @@ def single_exception_from_error_tuple(
     tb,  # type: Optional[TracebackType]
     client_options=None,  # type: Optional[Dict[str, Any]]
     mechanism=None,  # type: Optional[Dict[str, Any]]
+    exception_id=0,  # type: int
+    parent_id=0,  # type: int
+    source=None,  # type: Optional[str]
 ):
     # type: (...) -> Dict[str, Any]
-    mechanism = mechanism or {"type": "generic", "handled": True}
+    """
+    Creates a dict that goes into the events `exception.values` list and is ingestible by Sentry.
+
+    See the Exception Interface documentation for more details:
+    https://develop.sentry.dev/sdk/event-payloads/exception/
+    """
+    exception_value = {}  # type: Dict[str, Any]
+    exception_value["mechanism"] = (
+        mechanism.copy() if mechanism else {"type": "generic", "handled": True}
+    )
+    exception_value["mechanism"]["exception_id"] = exception_id
 
     if exc_value is not None:
         errno = get_errno(exc_value)
     else:
         errno = None
 
     if errno is not None:
-        mechanism.setdefault("meta", {}).setdefault("errno", {}).setdefault(
-            "number", errno
-        )
+        exception_value["mechanism"].setdefault("meta", {}).setdefault(
+            "errno", {}
+        ).setdefault("number", errno)
+
+    if source is not None:
+        exception_value["mechanism"]["source"] = source
+
+    is_root_exception = exception_id == 0
+    if not is_root_exception:
+        exception_value["mechanism"]["parent_id"] = parent_id
+        del exception_value["mechanism"]["handled"]
+
+        exception_value["mechanism"]["type"] = "chained"
+
+    if is_root_exception and "type" not in exception_value["mechanism"]:
+        exception_value["mechanism"]["type"] = "generic"
+
+    if exc_type == ExceptionGroup:
+        exception_value["mechanism"]["is_exception_group"] = True
+
+    exception_value["module"] = get_type_module(exc_type)
+    exception_value["type"] = get_type_name(exc_type)
+    exception_value["value"] = (
```

Comment:
before this was `safe_str(exc_value)` without any conditions, this doesn't seem to be the same logic as before.
Please explain what you're doing here.

Reply:
In case of an ExceptionGroup they have a message and it will use this message instead of the value because it is nicer to read. See also https://github.com/getsentry/rfcs/blob/main/text/0079-exception-groups.md#exception-value

Reply:
I still think this is a confusing way of writing this condition. 
If the requirement is to take the `exc_value.message` if it exists, I'd write it the other way as below so that it's clear when the older behavior is preserved. The way you wrote it, if `exc_value` is false-y it will go to `exc_value.message` and throw an error.

```python
    exception_value["value"] = (
         exc_value.message
         if exc_value and hasattr(exc_value, "message")
         else safe_str(exc_value)
     )
```

You can even simplify to 
```python
getattr(exc_value, "message", safe_str(exc_value))
```
which does the same as above.

Reply:
Nice. way cleaner!